tardis/energy_input/gamma_ray_channel.py

```diff
@@ -3,13 +3,64 @@
 import pandas as pd
 import astropy.units as u
 import radioactivedecay as rd
+from tardis.model.matter.decay import IsotopicMassFraction
 
 from tardis.energy_input.util import KEV2ERG
 
 logger = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
 
+def evolve_mass_fraction(
+    raw_isotope_mass_fraction,
+    time_start,
+    time_end,
+    time_step,
+    model_isotope_time,
+):
+    """
+    Function to evolve the isotope abundance over time.
+
+    Parameters
+    ----------
+    raw_isotope_mass_fraction : pd.DataFrame
+        mass fractions of isotopes.
+    time_start : float
+        Start time of simulation in days.
+    time_end : float
+        End time of simulation in days.
+    time_step : float
+        Time step of simulation in days.
+    model_isotope_time : float
+        isotope time coming from HESMA model (typically 100 s) in days.
+
+    Returns
+    -------
+    isotope_mass_fraction : pd.DataFrame
+        isotope mass fractions at different times including the initial time.
+    """
+
+    # linear time spacing
+    time_delta = np.linspace(time_start, time_end, time_step)
```

Comment:
Instead of taking in the start/end/step, please use whatever time array is passed in (so that we can support log-space time for example)

Reply:
Changed!